### PR TITLE
feat: UI + Logic Improvements

### PR DESCRIPTION
**Pull request by Bran18**
## Summary by Sourcery

Improve the user interface and logic of the authentication forms by adding password visibility toggles and a password strength meter. Refactor email templates into a separate module for better code organization. Enhance the responsiveness and animations of authentication-related pages. Simplify and improve the ClickableText component logic and truncate long usernames in the user menu for better UI consistency.

New Features:
- Introduce password visibility toggle in the signup and signin forms to enhance user experience.
- Add password strength meter to the signup form to guide users in creating strong passwords.

Enhancements:
- Refactor email templates by extracting them into a separate module for better maintainability and reusability.
- Improve UI responsiveness and add animations to the signup, signin, reset password, and forgot password pages using Framer Motion.
- Enhance the ClickableText component by simplifying its logic and improving text parsing capabilities.
- Truncate long usernames in the user menu to maintain UI consistency.

## Release Notes

- **New Features**
	- Enhanced user experience with animated transitions on the Forgot Password, Reset Password, Sign In, and Sign Up pages.
	- Added password visibility toggle in Sign In and Sign Up forms for improved usability.
	- Introduced a Password Strength Meter to guide users in creating strong passwords.
	- Implemented a new email template generation system for password reset and verification emails.
	- Added a new function to truncate long usernames in the User Menu for better readability.

- **Improvements**
	- Updated email handling to utilize dedicated functions for generating email templates, improving modularity.
	- Improved scroll management in the Browse Accordion component for mobile users.
	- Refined layout and responsiveness across various components, including adjustments to spacing and structure in BrowseChatbotDetails and BrowseUserDetails.
	- Streamlined the ChatPanel component for clearer disabled state conditions.
	- Enhanced the structure and styling of the ThreadPopup and ChatbotDetails components for better readability and user experience.
	- Updated the Sign In and Sign Up forms to improve layout and responsiveness.
	- Enhanced clickable text handling in the Chat component for better accessibility.

- **Bug Fixes**
	- Streamlined error handling and validation processes across various forms.

- **Documentation**
	- Added new interfaces and functions for better structure and clarity in email and password management.

### Previews: 

Screenshot 2024-11-07 at 11 12 02 AM
Screenshot 2024-11-07 at 11 12 31 AM
Screenshot 2024-11-07 at 11 12 51 AM
Screenshot 2024-11-07 at 11 13 31 AM
Screenshot 2024-11-07 at 11 14 07 AM
Screenshot 2024-11-07 at 11 15 58 AM
Screenshot 2024-11-07 at 11 16 42 AM
Screenshot 2024-11-07 at 11 17 34 AM
Screenshot 2024-11-07 at 1 14 58 PM
Screenshot 2024-11-07 at 1 15 34 PM
Screenshot 2024-11-07 at 1 23 17 PM
Screenshot 2024-11-07 at 1 51 23 PM

### CODE REVIEW

File: apps/masterbots.ai/components/routes/chat/chat-combobox.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ import { AIModels } from '@/app/api/chat/models/models'
 
 //* Model options available in the combobox, each with label, value, and logo icon.
 const models = [
-  { label: 'GPT-4o', value: AIModels.Default, logo: <IconOpenAI /> },
+  { label: 'GPT-4o', value: AIModels.Default, logo: "MB" },
```
[the remark this answers is not in the document]
Comment (Bran18) [reply apps/masterbots.ai/components/routes/chat/chat-combobox.tsx:33]: This is temporary, we are missing the MB word small icon. Once the logo is created we are going to be replacing the `MB` word with the original MB Icon

File: apps/masterbots.ai/components/routes/browse/browse-user-details.tsx [hunk 2]
```diff
@@ -30,35 +33,55 @@ export default function BrowseChatbotDetails({ user }: { user?: User | null }) {
...
+        </a>
 
-          <div className="text-base">
-            <div className="font-light">
-              Threads: <span className="text-[#71717A]">{threadNum ?? 1}</span>
+        <div className="dark:bg-[#09090B] bg-white rounded-lg p-6 relative font-mono">
+          <div className="flex flex-col gap-3">
+            <div className="text-2xl font-black">
+              {user?.username?.replace('_', ' ')}
+            </div>
```
[the remark this answers is not in the document]
Comment (Bran18) [reply apps/masterbots.ai/components/routes/browse/browse-user-details.tsx:67]: @coderabbitai I'm looking for a way to truncate and display the 3 dots only in LG/XL without having to create a custom reactive hook. In the meantime this solution is ok, and I'm 100% sure not to include the 3 dots for responsive reasons for now.

**Comment by Bran18** [comment]
### OLD V: 

https://github.com/user-attachments/assets/4483a318-99cf-4c7c-a6a8-6fcea5b2f9a3

**Review by AndlerRL — APPROVED** [review]
LGTM! Thanks for the screens reports comparing the before and after and taking care of the details in the mobile experience. I noticed the improvement also on the current bot profile card and the redirection back to browse :+1:

Merging for tests!

@AnoukRImola -- please you can go ahead and start testing the changes. Most of them are mobile experience improvements and other UI/UX improvements, so as the pw reset and setup.

@sheriffjimoh -- here are changes of your interest 👀 I noticed that on the bot profile card there is no "return to browser" link on the hero. Please keep that link, I missed to add that link into the designs in figma.

**Comment by AndlerRL** [comment]
[Brandon Development Support - November 2024, 1 of 2](https://trello.com/c/uQypMx4x/264-brandon-development-support-november-2024-1-of-2)

**Pull request merged**